hot-desk/src/main/java/com/exadel/sandbox/security/controller/AuthController.java

```diff
@@ -0,0 +1,33 @@
+package com.exadel.sandbox.security.controller;
+
+import com.exadel.sandbox.security.model.ApiResponse;
+import com.exadel.sandbox.security.model.LoginDto;
+import com.exadel.sandbox.security.model.UserRegisterDto;
+import com.exadel.sandbox.security.service.UserDetailServiceImp;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/auth")
+@RequiredArgsConstructor
+public class AuthController {
+    private final UserDetailServiceImp userService;
+
+    @PostMapping("register")
+    public ResponseEntity<?> registerUser(@RequestBody UserRegisterDto userRegisterDto) {
+        ApiResponse apiResponse = userService.registerUser(userRegisterDto);
+        return ResponseEntity.status(apiResponse.isSuccess() ? 201 : 409).body(apiResponse);
```

Comment:
It bad approach u throw an exception and handle it to get 409 response

Reply:
Have no idea. Do you have any better suggestions?